### PR TITLE
PFEGammaProducer with fillDescriptions

### PR DESCRIPTION
**Pull request by guitargeek**
Continuing cleaning up the PFEGammaProducer/Algo.

This time, I moved the configuration to the fillDescriptions function and removed unused parameters on that occasion.

Local matrix tests pass.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoParticleFlow/PFProducer/plugins/PFEGammaProducer.cc [hunk 1]
```diff
@@ -644,3 +506,65 @@ void PFEGammaProducer::createSingleLegConversions(reco::PFCandidateEGammaExtraCo
...
+  desc.add<double>         ("sumEtEcalIsoForEgammaSC_endcap",   2.);
+  desc.add<double>         ("coneEcalIsoForEgammaSC",           0.3);
+  desc.add<double>         ("sumPtTrackIsoForEgammaSC_barrel",  4.);
+  desc.add<double>         ("sumPtTrackIsoForEgammaSC_endcap",  4.);
+  desc.add<double>         ("coneTrackIsoForEgammaSC",          0.3);
+  desc.add<unsigned int>   ("nTrackIsoForEgammaSC",             2);
+  desc.add<double>         ("pf_electron_mvaCut",               -0.1);
+  desc.add<bool>           ("pf_electronID_crackCorrection",    false);
+  desc.add<double>         ("pf_conv_mvaCut",                   0.0);
+  desc.add<double>         ("sumPtTrackIsoForPhoton",           2.0);
```
Comment (perrotta) [inline RecoParticleFlow/PFProducer/plugins/PFEGammaProducer.cc]: This is for the PFPhotonAlgo: since you are cleaning, is it really needed here?

**Review by perrotta — COMMENTED** [review]
Please also consider adding comments to (some of) the parameters inserted with your fillDescription method (see https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp#Adding_Comments) so that the auto-produced cfi will become more easily readable

File: RecoParticleFlow/PFProducer/plugins/PFEGammaProducer.cc [hunk 2]
```diff
@@ -25,81 +25,35 @@
...
-    =  iConfig.getParameter<bool>("usePhotonReg");
-
-  useVerticesForNeutral_
-    = iConfig.getParameter<bool>("useVerticesForNeutral"); 
-
-  useRegressionFromDB_
-    = iConfig.getParameter<bool>("useRegressionFromDB"); 
-
-
   std::vector<double>  calibPFSCEle_Fbrem_barrel;
```
Comment (perrotta) [inline RecoParticleFlow/PFProducer/plugins/PFEGammaProducer.cc]: Also the definitions, then...

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- PFEGammaProducer cleaned from unused class members and configuration parameters; fillDescription() method is added to deal with those parameters
- Said so, the pull request is kind of technical: no effects are visible in the standard outputs, and none were expected

**Comment by fabiocos** [comment]
+1

**Pull request merged**